pkg/rancher-desktop/main/update/LonghornProvider.ts

```diff
@@ -380,9 +373,15 @@ export default class LonghornProvider extends Provider<LonghornUpdateInfo> {
 
     const queryResult = await queryUpgradeResponder(this.configuration.upgradeServer, this.updater.currentVersion);
     const { latest, unsupportedUpdateAvailable } = queryResult;
-    const requestIntervalInMinutes = queryResult.requestIntervalInMinutes || defaultUpdateIntervalInMinutes;
-    const requestIntervalInMs = requestIntervalInMinutes * 1000 * 60;
-    const nextRequestTime = Date.now() + requestIntervalInMs;
+
+    // Trigger the next check no earlier than some early time in the next calendar day.
+    const nextRequestTime = new Date();
```

Comment:
I'm wondering how we handle potential calendar-day edge cases here. Should we compute the start of the day first and then add the jitter in milliseconds?

Reply:
We add one day on the next line.

So if it's currently Nov 6, 23:59: After the next two lines, the next trigger is between Nov 7 00:00 ~ 01:59:…
If it's currently Nov 7, 00:00: the next trigger is Nov 8, 00:00 ~ 01:59:…

(I re-read the spec, `Math.random()` returns in the range `[0, 1)`.)